### PR TITLE
Feat: Retrieve And Show Patient List

### PR DESCRIPTION
**Pull request by pablopalou**
image

### CODE REVIEW

File: patient-data-management/src/components/PatientCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import { IPatient } from '../../types/Patient';
+
+interface PatientCardProps {
+  patient: IPatient;
+}
```
Comment (ThunderNaka) [inline patient-data-management/src/components/PatientCard/index.tsx:6]: If you just have one property, do it inline xd

File: patient-data-management/src/components/PatientList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const PatientList: React.FC = () => {
+    const { patients, isLoading, error } = usePatients();
+
+    if (isLoading) {
+        return <div>Loading...</div>;
+    }
+
+    if (error) {
+        const errorMessage = (error as Error).message || 'An error occurred';
```
Comment (ThunderNaka) [inline patient-data-management/src/components/PatientList/index.tsx:12]: Try to avoid casting.

File: patient-data-management/src/utils/api.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios';
```
Comment (ThunderNaka) [inline patient-data-management/src/utils/api.ts:1]: You are importing axios but you don't have it in the package.json 🤔

File: patient-data-management/package.json [hunk 4]
```diff
@@ -11,6 +11,7 @@
     "@types/react": "^18.2.31",
     "@types/react-dom": "^18.2.14",
     "react": "^18.2.0",
+    "react-axios": "^2.0.6",
```
Comment (ThunderNaka) [inline patient-data-management/package.json:14]: I think the library we use is axios, not react-axios, but not sure why is this working for you 🤔

File: patient-data-management/src/App.tsx [hunk 5]
```diff
@@ -1,14 +1,21 @@
 import './App.css';
+import PatientList from './components/PatientList';
+import { QueryClient, QueryClientProvider } from 'react-query';
 
 function App() {
+  const queryClient = new QueryClient();
```
Comment (ThunderNaka) [inline patient-data-management/src/App.tsx:6]: You can declare this outside the main component.

File: patient-data-management/src/App.tsx [hunk 6]
```diff
@@ -1,14 +1,21 @@
...
-        <h1 className="text-xl font-bold underline">
-          Hello world!
+    <QueryClientProvider client={queryClient}>
+      <div className="App">
+        <h1 className="text-3xl font-semibold mt-8 mb-4">
+          Patient list:
         </h1>
-      </header>
-    </div>
+        <div>
```
Comment (ThunderNaka) [inline patient-data-management/src/App.tsx:14]: Is this extra div needed? 🤔

**Review by ThunderNaka — APPROVED** [review]
Great PR overall! A few things to fix but keep up the good work! 💪

File: patient-data-management/src/components/PatientList/index.tsx (hunk 2 above)
Comment (pablopalou) [reply patient-data-management/src/components/PatientList/index.tsx:12]: How would you do this? Typescript was complaining

**Pull request merged**